### ISSUE TITLE
Regex wishlist/idealist

### ISSUE THREAD

**Issue by NilsEnevoldsen** [scryfall/feedback#808]
- [ ] Mana symbol wildcard (`::sm::`: colored, colorless, generic, phyrexian, hybrid, twobrid, snow, X/Y/Z, halves, hundred, million, infinity, …?)
  - [ ] Variant: Symbols wildcard (`::s::`: `::sm::`+ tapping, untapping, loyalty, chaos, energy, etc)
  - [ ] Variant: Mana "class" wildcards (`::phyrexian::`, `::hybrid::`, etc)
- [ ] Single/multiline mode toggle (`o:/flying*scry/` vs `o:/flying*scry/m`)
  - What default?
- [ ] Single character wildcard (`.`)
- [ ] Zero-or-more token repetitions (`*`)
- [ ] One-or-more token repetitions (`+`) (Current `+` only works with certain tokens)
- [ ] Whitespace wildcard (`\s`), non-whitespace wildcard (`\S`)
- [ ] Card-field name wildcards other than `~` (`::subtype::`) (Gatherer feature parity) (Maybe don't match `::subtype::` to an instance of `~`, though?)
- [ ] Type wildcards (`::creaturesubtype::` to match any creature subtype)
- [ ] Backreferences

**Comment by NilsEnevoldsen** [comment]
BTW, I encourage anybody who sees this issue to "vote" for what they consider important. I don't think Scryfall devs are eager to drag themselves over the hot coals of regex hell to implement features that nobody will use.

Personally, I'd like single-line mode, a mana symbol wildcard, and a more general one-or-more token operator.

**Comment by spacemonaut** [comment]
All of these I can take or leave right now since I haven't run into a situation where I've needed one.

`::subtype::` as such doesn't seem useful to me. If I care about a tribe, it's almost certainly because I'm building a tribal deck. If I'm building knights I'm going to search `knight` specifically. I won't search `t:knight o:/::subtype::/` because I want to know about non-knight sources that might care about knights. If I wanted knights that care about knights, `t:knight o:knight` would be preferable so that I don't instead capture a goblin knight that cares about goblins.

**Comment by gmac** [comment]
Corey can speak to this the best, but I'll offer a short summary: Scryfall regex capabilities have been carefully selected to account for security and performance. If you're familiar with regex, you've probably heard people talk about the [inherent risks](http://stackstatus.net/post/147710624694/outage-postmortem-july-20-2016) that they pose. It's very easy to write bad regex due to inexperience, and bad regex have an innate ability to bog down applications.

Long story short, I would hold your breath waiting for more advanced features.

**Comment by csuhta** [comment]
Regular expression support has been vastly improved, is now PCRE-compatible:
https://scryfall.com/docs/regular-expressions
